src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -833,6 +833,11 @@ private void Get_CORINFO_SIG_INFO(MethodDesc method, CORINFO_SIG_INFO* sig, Meth
         {
             Get_CORINFO_SIG_INFO(method.Signature, sig, scope);
 
+            // Get_CORINFO_SIG_INFO above will have correctly set CORINFO_CALLCONV_ASYNCCALL for async variants,
+            // however the special async intrinsics don't set the async bit in their signature. Compensate for that.
+            if (method.IsAsync && !method.Signature.ReturnsTaskOrValueTask())
```

Comment:
```suggestion
            if (method.IsAsyncIntrinsic())
```
?

and create extension method for this